tests/drivers/gpio/gpio_enable_disable_interrupt/src/main.c

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2022 Google Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/gpio/gpio_emul.h>
+#include <zephyr/kernel.h>
+#include <zephyr/ztest.h>
+
+#define DEV    DT_GPIO_CTLR(DT_INST(0, test_gpio_enable_disable_interrupt), in_gpios)
+#define PIN_IN DT_GPIO_PIN(DT_INST(0, test_gpio_enable_disable_interrupt), in_gpios)
+
+static struct gpio_callback cb_data;
+static bool cb_called;
+
+static void callback(const struct device *dev, struct gpio_callback *gpio_cb, uint32_t pins)
+{
+	cb_called = true;
+}
+
+struct gpio_enable_disable_interrupt_fixture {
+	const struct device *port;
+	gpio_pin_t pin;
+};
+
+static void *gpio_enable_disable_interrupt_setup(void)
+{
+	static struct gpio_enable_disable_interrupt_fixture fixture;
+
+	fixture.pin = PIN_IN;
+	fixture.port = DEVICE_DT_GET(DEV);
+
+	return &fixture;
+}
+
+static void gpio_enable_disable_interrupt_before(void *arg)
+{
+	struct gpio_enable_disable_interrupt_fixture *fixture =
+		(struct gpio_enable_disable_interrupt_fixture *)arg;
+
+	zassert_true(device_is_ready(fixture->port), "GPIO device is not ready");
+
+	zassert_ok(gpio_pin_configure(fixture->port, fixture->pin, GPIO_INPUT));
+	zassert_ok(gpio_emul_input_set(fixture->port, fixture->pin, 0),
+		   "failed to set value on input pin");
+	cb_called = false;
+
+	zassert_ok(gpio_pin_interrupt_configure(fixture->port, fixture->pin, GPIO_INT_DISABLE));
+	gpio_init_callback(&cb_data, callback, BIT(fixture->pin));
+	zassert_ok(gpio_add_callback(fixture->port, &cb_data), "failed to add callback");
```

Comment:
You should create an "after" function that calls `gpio_remove_callback()`.

Reply:
Thank you for catching this issue, fixed.